accepted/0000-sle-micro-reboot-needed.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: SLE Micro: report when a reboot is needed
+- Start Date: 2021-09-23
+
+# Summary
+[summary]: #summary
+
+After a system level change in SLE Micro, SUSE Manager should indicate in the UI when a reboot is needed.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, for SLE Micro, we display the following message on the detail page: `OS with transactional updates: please reboot after any packages and/or channel changes`.
+
+This is not ideal when it comes to usability. The user is responsible for knowing when a reboot is needed, and scheduling those reboots.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The SLE Micro uses transactional updates. Transactional updates are atomic (all updates are applied only if all updates succeed) and support rollbacks. They don't affect a running system as no changes are activated until after the system is rebooted.
+To be able to have up-to-date information regarding reboot needed, we need to update this information after every state apply.
+For that, this proposed solution is based on the following steps:
+ - Create an execution module to check if a reboot is required
```

Comment:
WRT grains: sadly I did not added the grain in the [extra](https://github.com/saltstack/salt/commit/94b6ab1cdbe78513e066fc95c70039ff727c8674) grains, but luckily there is a [place](https://github.com/saltstack/salt/blob/master/salt/grains/pending_reboot.py) to add one upstream!!

Reply:
> No, grains do not update automatically. Grains are made for values which do not change often. It will not work for this usecase.

Grains can be refreshed, and this is the approach the upstream is using for Windows (see last comment)

Reply:
but refreshing grains after every change can be very expensive. You refresh "all" grains, not only the one which you are interested in.
Additionally I see the following happen.
- user  update a kernel package
- user request reboot needed grain
  -> no reboot needed

Why? Forgot to refresh. The grains was taken out of the cache and that a kernel was updated did not refresh the grain.